scripts/provision_credentials.py

```diff
@@ -143,6 +156,22 @@ def _destroy_service_account_creds(self, service_account_email):
             logger.info("Successfully deleted service account key with id '%s' for user '%s'",
                         key_id, service_account_email)
 
+    def _register_sam(self, google_key):
+        pool = urllib3.PoolManager(ca_certs=certifi.where())
+        credentials = service_account.Credentials.from_service_account_info(google_key)
+        token = self._get_access_token(credentials)
+        response = pool.request('POST',
+                                self.sam_endpoint,
+                                body='',
+                                headers={'Authorization': f'Bearer {token}'})
+        if response.status != 200:
```

Comment:
It looks like SAM returns a 201 in the success case which would trigger a false error.

That seems to indicate that this wasn't tested properly.

Reply:
For idempotency, the expected error would be 409.

Reply:
I guess it's hard to test without first deregistering the SA. Does SAM have the ability to deregister a user?

Reply:
Not as far as I can tell. https://sam.dsde-prod.broadinstitute.org/

Reply:
Better URL: https://sam.dsde-prod.broadinstitute.org/#/Users